src/main/java/org/folio/entitlement/configuration/RetryConfiguration.java

```diff
@@ -0,0 +1,88 @@
+package org.folio.entitlement.configuration;
+
+import jakarta.ws.rs.WebApplicationException;
+import java.util.function.Predicate;
+import org.folio.entitlement.integration.IntegrationException;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.backoff.BackOffPolicyBuilder;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+@EnableRetry
+@Configuration
+public class RetryConfiguration {
+
+  @Value("${retries.module.max:3}") private int maxRetriesForFolioModuleCalls = 3;
```

Comment:
The advantage is that search by config property name (e.g. search for "retries.module.max") will match the code, while with @ConfigurationProperties it will not.

But I will change the code to `@ConfigurationProperties` since this is the common practice here.